api/src/main/java/com/example/api/service/validator/BadgeValidator.java

```diff
@@ -0,0 +1,49 @@
+package com.example.api.service.validator;
+
+import com.example.api.dto.request.user.badge.BadgeForm;
+import com.example.api.error.exception.EntityNotFoundException;
+import com.example.api.error.exception.ExceptionMessage;
+import com.example.api.error.exception.RequestValidationException;
+import com.example.api.model.user.badge.Badge;
+import com.example.api.util.MessageManager;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+@Component
+@Slf4j
+@RequiredArgsConstructor
+public class BadgeValidator {
+
+    public void validateBadgeIsNotNull(Badge badge, Long id) throws EntityNotFoundException {
+        if (badge == null) {
+            log.error("Badge with id {} does not exist", id);
+            throw new EntityNotFoundException("Badge with id " + id + " does not exist");
+        }
+    }
+
+    public Integer validateAndGetIntegerValue(String value) throws RequestValidationException {
+        try {
+            return Integer.valueOf(value);
+        } catch (NumberFormatException e) {
+            log.error("String value should be parsable to int!");
+            throw new RequestValidationException(ExceptionMessage.STRING_VALUE_NOT_PARSABLE_TO_INT);
+        }
+    }
+
+    public void validateBadgeForm(BadgeForm form) throws RequestValidationException {
+        if (form.getTitle().length() > 30) {
+            log.error("Title should be max 32 characters long!");
```

Comment:
No to 30 czy 32 ? 😛

Reply:
xdddd mialem 32 ale zmienilem potem na 30 bo gdzies tam w innym miejsu jest 30